src/tools/compiletest/src/runtest.rs

```diff
@@ -1725,6 +1725,28 @@ impl<'test> TestCx<'test> {
         }
     }
 
+    fn is_vxworks_pure_static(&self) -> bool {
+        if self.config.target.contains("vxworks") {
+            match env::var("RUST_TEST_DYLINK") {
+                Ok(s) => s != "1",
+                _ => true
+            }
+        } else {
+            false
+        }
+    }
+
+    fn is_vxworks_pure_dynamic(&self) -> bool {
+        if self.config.target.contains("vxworks") {
+            match env::var("RUST_TEST_DYLINK") {
```

Comment:
Could this branch perhaps just be `!self.is_vxworks_pure_static()`? (basically reading the env var only once)

Reply:
> Could this branch perhaps just be `!self.is_vxworks_pure_static()`? (basically reading the env var only once)

For !self.is_vxworks_pure_static(), it would be true for all non-vxworks targets which is not what we want.

Reply:
Er sorry I meant just in this branch, so only if the target is vxworks do we delegate to `is_vxworks_pure_static`

Reply:
I see. Please review the latest code.